Library/Homebrew/cmd/install.rb

```diff
@@ -176,6 +180,12 @@ class InstallCmd < AbstractCommand
 
       sig { override.void }
       def run
+        # Apply non-interactive behaviour for sudo/interactive prompts when requested
+        ENV["HOMEBREW_NON_INTERACTIVE"] = "1" if args.value("non-interactive-sudo").present?
+        if (timeout = args.value("timeout-wait-for-user")).present?
+          ENV["HOMEBREW_PROMPT_TIMEOUT_SECS"] = timeout
+        end
```

Comment:
Avoid using environment variables like this to set global state. Instead pass `args.non_interactive_sudo` and `args.timeout_wait_for_user` values (shouldn't need to use `args.value` if you use `brew typecheck --update` or adjust the `env_config.rbi` files manually) to `Cask::Installer`.

If you need to set something globally on SystemCommand consider a `SystemCommand.non_interactive_sudo!` method or similar that's called conditionally and sets class-wide state there instead.